brave/src/main/java/brave/internal/recorder/PendingSpans.java

```diff
@@ -58,6 +58,11 @@ public PendingSpan getOrCreate(TraceContext context, boolean start) {
     PendingSpan newSpan = new PendingSpan(data, clock);
     PendingSpan previousSpan = delegate.putIfAbsent(new RealKey(context, this), newSpan);
     if (previousSpan != null) return previousSpan; // lost race
+
+    if (Platform.get().logEnabled()) {
+      newSpan.caller =
+          new Throwable("Thread " + Thread.currentThread().getName() + " allocated span here");
```

Comment:
Couldn't this be done in the constructor of `PendingSpan`? If so, the `caller` field could be `final`.

Reply:
if we did, on putIfAbsent race loss we would have unnecessary allocated a throwable. however, maybe that is ok if we consider performance will be hit on debug anyway.. wdyt?